token/program-2022/src/state.rs

```diff
@@ -80,6 +127,53 @@ impl Pack for Mint {
     }
 }
 
+/// Type-Length-Value Entry, used to encapsulate all mixins contained within an account
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct TlvEntry<T, V> {
+    pub account_type: T,
+    pub length: u64, // usize? or u32?
+    pub value: V,
+}
+
+/// Close account mixin data for mints.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct MintCloseMixin {
+    /// Optional authority to close the mint
+    pub close_authority: COption<Pubkey>,
+}
+
+/// Transfer fee information
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct TransferFee {
+    /// First epoch where the transfer fee takes effect
+    pub epoch: Epoch,
+    /// Amount of transfer collected as fees, expressed as basis points of the
+    /// transfer amount, ie.  increments of 0.01%
+    pub transfer_fee_basis_points: u16,
+    /// Maximum fee assessed on transfers, expressed as an amount of tokens
+    pub maximum_fee: u64,
+}
+
+/// Transfer fee mixin data for mints.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct MintTransferFeeMixin {
+    /// Optional authority to set the fee
+    pub fee_config_authority: COption<Pubkey>,
+    /// Any harvest instruction must be signed by this key. If set to `None`, then
+    /// this mint behaves like a normal `Mint`.
+    pub fee_withdraw_authority: COption<Pubkey>,
+    /// Withheld transfer fee tokens that have been moved to the mint for harvesting
+    pub withheld_amount: u64,
+    /// Old transfer fee, used if the current epoch < new_transfer_fee.epoch
+    pub old_transfer_fee: TransferFee,
+    /// New transfer fee, used if the current epoch >= new_transfer_fee.epoch
+    pub new_transfer_fee: TransferFee,
```

Comment:
I had that initially, and then thought it would be annoying to have `transfer_fee[0]` and `transfer_fee[1]` all over the place, when it's totally clear which one is newer and older with explicit names.  Maybe we just need a better name? `older_transfer_fee` and `newer_transfer_fee`? `current_transfer_fee` and `next_transfer_fee`? `current` and `upcoming`? `now` and `later`? I don't like any of these, but `older` and `newer` may be clearest